Fastly/fastly/connector_fastly_waf_base.py

```diff
@@ -0,0 +1,305 @@
+                self.log(
+                    message=f"{self.name}: No events to forward",
+                    level="info",
+                )
+
+        # get the ending time and compute the duration to fetch the events
+        batch_end_time = time.time()
+        batch_duration = int(batch_end_time - batch_start_time)
+        self.log(
+            message=f"{self.name}: Fetched and forwarded events in {batch_duration} seconds",
+            level="debug",
+        )
+        FORWARD_EVENTS_DURATION.labels(intake_key=self.connector.configuration.intake_key).observe(batch_duration)
+
+        # compute the remaining sleeping time. If greater than 0, sleep
+        delta_sleep = self.connector.configuration.frequency - batch_duration
+        if delta_sleep > 0:
+            self.log(
+                message=f"{self.name}: Next batch in the future. Waiting {delta_sleep} seconds",
+                level="debug",
+            )
+            time.sleep(delta_sleep)
+
+    def run(self) -> None:
+        while self.running:
+            try:
+                self.next_batch()
+            except Exception as error:
+                self.log_exception(error, message=f"{self.name}: Failed to forward events")
+
+
+class FastlyWAFBaseConnector(Connector):
+    configuration: FastlyWAFBasicConnectorConfiguration
+    base_uri = "https://dashboard.signalsciences.net"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.context_lock = Lock()
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(email=self.configuration.email, token=self.configuration.token)
+
+    def get_sites(self):
+        result = []
+
+        page_num = 1
+        page_limit = 100
+        while True:
+            response = self.client.get(
+                f"{self.base_uri}/api/v0/corps/{self.configuration.corp}/sites?" f"page={page_num}&limit={page_limit}",
+                timeout=60,
+            )
+            response.raise_for_status()
+
+            site_names = [site["name"] for site in response.json().get("data", [])]
+            result.extend(site_names)
+
+            if len(site_names) < page_limit:
+                break
+
+            page_num += 1
+
+        return result
+
+    def start_consumers(self):
+        consumers = {}
+
+        # Always add endpoint for the corp, if available
+        corp_url = self.get_url_for_corp()
+        if corp_url:
+            consumers[f"corp:{self.configuration.corp}"] = FastlyWAFConsumer(
+                connector=self, name=f"corp:{self.configuration.corp}", url=corp_url
+            )
+
+        # If site is provided - use it
+        # If not - get all sites in the corp
+        if self.configuration.site is not None:
+            sites = [self.configuration.site]
+
+        else:
+            sites = self.get_sites()
+            print("sites", sites)
```

Comment:
I suggest removing this line or maybe define a log that display the count of discovered sites:
```suggestion
```